crates/telio-starcast/src/transport.rs

```diff
@@ -0,0 +1,243 @@
+#![allow(missing_docs)] // TODO: Remove
+
+use std::{
+    collections::{hash_map::DefaultHasher, HashMap},
+    hash::{Hash, Hasher},
+    net::{IpAddr, Ipv4Addr, SocketAddr},
+    sync::Arc,
+};
+
+use async_trait::async_trait;
+use futures::future::join_all;
+use ipnetwork::IpNetwork;
+use pnet_packet::{
+    ipv4::{checksum, MutableIpv4Packet},
+    udp::MutableUdpPacket,
+};
+use tokio::{net::UdpSocket, sync::RwLock};
+
+use telio_sockets::SocketPool;
+use telio_task::{io::Chan, task_exec, Runtime, RuntimeExt, Task, WaitResponse};
+use telio_utils::{telio_log_debug, telio_log_error, telio_log_warn};
+
+#[cfg(feature = "test-util")]
+const MULTICAST_TRANSPORT_PORT: u16 = 0;
+#[cfg(not(feature = "test-util"))]
+const MULTICAST_TRANSPORT_PORT: u16 = 13569;
+const MAX_PACKET_SIZE: usize = u16::MAX as usize;
+const IPV4_HEADER: usize = 20; // bytes
+const UDP_DEST_OFFSET: usize = 2;
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error("Could not bind to socket")]
+    SocketBindError(String),
+    /// Task execution failed
+    #[error(transparent)]
+    Task(#[from] telio_task::ExecError),
+}
+
+pub struct Config {
+    peers: Vec<IpAddr>,
+    multicast_ips: Vec<IpNetwork>,
```

Comment:
I think this can be moved, to  start as an arg, as it will not change after setup